src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -289,6 +289,8 @@ void Server::unregisterPath(const u16string& path) {
     }
     auto& watchPoint = it->second;
     watchPoint.cancel();
+    // Make sure we handle the cancellation
```

Comment:
```suggestion
    // Make sure the status of the watch point transitions to FINISHED
```

Or something along those lines would be better I think.

Can we be sure that the event already did arrive when we call `processQueues`?

Reply:
I chose a different path. PTAL.